tests/test_parsers.py

```diff
@@ -0,0 +1,136 @@
+"""
+Tests for ralph.parsers module.
+"""
+import gzip
+import logging
+import shutil
+
+import pandas as pd
+import pytest
+
+from ralph import filters
+from ralph.parsers import BaseParser, GELFParser
+
+
+def test_baseparser_parse():
+    """Test that calling parse on the BaseParser raises a NotImplementedError."""
+
+    with pytest.raises(NotImplementedError):
+        BaseParser().parse("foo.log")
+
+    # pylint: disable=abstract-method
+    class MyParser(BaseParser):
+        """Dummy parser."""
+
+    with pytest.raises(NotImplementedError):
+        MyParser().parse("foo.log")
+
+
+# pylint: disable=invalid-name
+def test_gelfparser_parse_raw_file(fs, gelf_logger):
+    """Test the GELFParser parsing using a flat GELF file."""
+
+    parser = GELFParser()
+
+    # Input log file does not exists
+    with pytest.raises(OSError):
+        parser.parse("/i/do/not/exist")
+
+    # Input log file is empty
+    empty_file_path = "/var/log/empty"
+    fs.create_file(empty_file_path)
+    events = parser.parse(empty_file_path)
+    assert len(events) == 0
+    assert events.equals(pd.DataFrame())
+
+    # Create two log entries
+    gelf_logger.info('{"username": "foo"}')
+    gelf_logger.info('{"username": ""}')
+
+    events = parser.parse(gelf_logger.handlers[0].stream.name)
+    assert len(events) == 2
+    assert events.equals(pd.DataFrame({"username": ["foo", ""]}))
+
+
+# pylint: disable=invalid-name,unused-argument
+def test_gelfparser_parse_gzipped_file(fs, gelf_logger):
+    """Test the GELFParser parsing using a gzipped GELF file."""
+
+    gelf_logger.info('{"username": "foo"}')
+    gelf_logger.info('{"username": "boo"}')
```

Comment:
```suggestion
    gelf_logger.info('{"username": "bar"}')
```
please respect the `foo` `bar` `baz` convention 😁